xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -63,21 +63,62 @@ viewers.Comments = Class.create({
   },
   /** Enhance the Comments UI with JS behaviors. */
   startup : function () {
-    if ($("commentform")) {
-      this.form = $("commentform").up("form");
-    } else {
-      this.form = undefined;
-    }
     this.loadIDs();
     this.addReplyListener();
-    this.addSubmitListener(this.form);
-    this.addCancelListener();
     this.addEditListener();
-    this.reloadEditor({
-      callback: function () {
-        this.addPreview(this.form);
-      }.bind(this)
-    });
+    // replaces the comment button with the form on the first click.
+    $("OpenCommentForm").observe('click', function (event) {
+      event.stop();
+      this.toggleFormHidden(function () {
+        this.reloadEditor({
+          callback: function () {
+            this.getForm();
+            this.addSubmitListener(this.form);
+            this.addCancelListener();
+            this.addPreview(this.form);
+          }.bind(this)
+        });
+      }.bind(this));
+    }.bind(this));
+  },
+  // lazily load the form element
+  getForm: function () {
+    this.form = this.form || undefined;
+    var commentform = $("commentform");
+    if (this.form === undefined && commentform) {
+      this.form = commentform.up("form");
+    }
+  },
+  // the form must be loaded only once.
+  formToggled: false,
+  // load the form only once, when it is first needed
+  // the next calls to this method does nothing but calling the callback
+  // function parameter.
+  toggleFormHidden: function (callback) {
+    if (!this.formToggled) {
+      require(['jquery'], function ($) {
+        var url = window.location.href;
+        $.ajax({
+          type: 'POST',
+          url: url,
+          data: {
+            'xpage': 'xpart',
+            'vm': 'commentform.vm',
+            'name': 'XWiki.XWikiComments'
+          },
+          success: function (data, textStatus, request) {
+            var placeHolder = $("#CommentFormPlaceholder");
+            placeHolder.replaceWith(data);
+            var button = $("#OpenCommentForm");
+            button.remove();
+            this.formToggled = true;
+            callback();
```

Comment:
Using **promises** is nicer and cleaner than using callbacks, when working with asynchronous operations.